assets/src/dashboard/app/views/templateDetails/content/index.js

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { useEffect } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import {
+  TemplateActionsPropType,
+  TemplatesPropType,
+  PageSizePropType,
+  TemplatePropType,
+} from '../../../../types';
+import {
+  DetailViewContentGutter,
+  Layout,
+  useLayoutContext,
+} from '../../../../components';
+import DetailsGallery from './detailsGallery';
+import RelatedGrid from './relatedGrid';
+
+function Content({
+  activeTemplateIndex,
+  isRTL,
+  orderedTemplatesLength,
+  pageSize,
+  relatedTemplates,
+  switchToTemplateByOffset,
+  template,
+  templateActions,
+}) {
+  const {
+    actions: { scrollToTop },
+  } = useLayoutContext();
+
+  useEffect(() => {
+    if (template) {
+      scrollToTop();
+    }
+  }, [template, scrollToTop]);
```

Comment:
this is new :)

Reply:
So if something changes in the template itself, but not the template this could fire that. I'd like to scrollToTop if the template id changes. We can use a previous value ref to do this.

`const previousTemplateId = useRef(template?.id)`

```
useEffect(() => {
  if (template !== null && template?.id !== previousTemplateId.current) {
    scrollToTop();
    previousTemplateId.current = template.id
  }
}, [template, scrollToTop]);
```

Reply:
You could probably get away with not even using a ref

```javascript
const templateId = template?.id;
useEffect(() => {
  if (templateId) {
    scrollToTop();
  }
}, [templateId, scrollToTop]);
```

Reply:
good idea! Currently the only time the template changes is when you either select an arrow or a different template from the grid below but I like how robust this is, thanks!